src/main/java/io/github/cottonmc/staticdata/ItemImpl.java

```diff
@@ -0,0 +1,60 @@
+package io.github.cottonmc.staticdata;
+
+import net.minecraft.util.Identifier;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+final class ItemImpl implements StaticData.Item {
```

Comment:
Feels a bit weird to have this package-private but leave everything else public. Folks could implement their own `DataSource` and load static data from a new place. Probably not too much of a problem, honestly.

Reply:
As someone who was at the bottom of the idea:
The whole point of the mod is so you need not implement anything like that, and can use a data/resource pack like structure. 

The only stuff that is NOT covered right now and you may want to Implement it is namespaces in the external static data folder, and the usage of zip files as static data packs. If that is there then I honestly don't know what else we could require.